bchrpc/server.go

```diff
@@ -1064,19 +1082,31 @@ func (s *GrpcServer) SubscribeTransactions(req *pb.SubscribeTransactionsRequest,
 						continue
 					}
 
-					header := block.MsgBlock().Header
+					toSend := &pb.TransactionNotification{}
 
-					respTx := marshalTransaction(tx, s.chain.BestSnapshot().Height-block.Height(), &header, block.Height(), s.chainParams)
-					if s.txIndex != nil {
-						if err := s.setInputMetadata(respTx); err != nil {
-							return err
+					if serializeTx {
+						var buf bytes.Buffer
+						if err := tx.MsgTx().BchEncode(&buf, wire.ProtocolVersion, wire.BaseEncoding); err != nil {
+							return status.Error(codes.Internal, "error serializing transaction")
 						}
-					}
-					toSend := &pb.TransactionNotification{
-						Type: pb.TransactionNotification_CONFIRMED,
-						Transaction: &pb.TransactionNotification_ConfirmedTransaction{
+						toSend.Type = pb.TransactionNotification_CONFIRMED
+						toSend.Transaction = &pb.TransactionNotification_SerializedTransaction{
+							SerializedTransaction: buf.Bytes(),
+						}
+
+					} else { //Marshal tx
```

Comment:
NIT: don't think this comment is really needed, especially if we have helper methods with good names and comments for them!